core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -589,7 +593,10 @@ public void updateRetainedTransports(Collection<EquivalentAddressGroup> addrs) {
     @Override
     public ClientTransport getTransport(final EquivalentAddressGroup addressGroup) {
       checkNotNull(addressGroup, "addressGroup");
-      TransportSet ts;
+      TransportSet ts = transports.get(addressGroup);
```

Comment:
> I meant locking internal to `TransportSet ts` rather than `transports`

Oh, I see. Yes, it would pressure `TransportSet` more, but it already has a [lock-free fast path](https://github.com/grpc/grpc-java/blob/v1.0.0/core/src/main/java/io/grpc/internal/TransportSet.java#L173).